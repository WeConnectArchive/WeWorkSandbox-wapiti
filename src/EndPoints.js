import React, {Component} from 'react';
import {EndPoint} from './EndPoint';
  
export class EndPoints extends Component {

  constructor(props) {
    super(props);
  }

  render() {
    return (<React.Fragment>
    {this.props.paths
    .map(
      path =>  (<React.Fragment><span/><EndPoint
        selectedServer={this.props.selectedServer}
        path={path}
        specType={this.props.specType}
      > </EndPoint>
      </React.Fragment>)
    )}
  </React.Fragment>)
  }

}
