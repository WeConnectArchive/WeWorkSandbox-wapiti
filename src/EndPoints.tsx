import * as React from 'react';
import {EndPoint} from './EndPoint';
  
interface EndPointsProps {
    paths: any;
    selectedServer: string;
    specType: string;
  };

export class EndPoints extends React.Component<EndPointsProps> {

  render() {
    return (<>
    {this.props.paths
    .map(
      path =>  (<><span/><EndPoint
        selectedServer={this.props.selectedServer}
        path={path}
        specType={this.props.specType}
      > </EndPoint>
      </>)
    )}
  </>)
  }

}
