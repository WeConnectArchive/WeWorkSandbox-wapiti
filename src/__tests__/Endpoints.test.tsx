import * as React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { EndPoints } from '../EndPoints';

const thePath = [{"test1":"test1"},{"test2":"test2"}]
const theType = "rest"

describe('Endpoints', () => {
  it('renders with rest', () => {
    let wrapper = shallow(<EndPoints
              selectedServer  = ""
              paths           = {thePath}
              specType        = "rest"
            ></EndPoints>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
