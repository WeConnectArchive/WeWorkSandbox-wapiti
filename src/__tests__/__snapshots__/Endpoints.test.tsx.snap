// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Endpoints renders with rest 1`] = `
<Fragment>
  <span />
  <EndPoint
    path={
      Object {
        "test1": "test1",
      }
    }
    selectedServer=""
    specType="rest"
  >
     
  </EndPoint>
  <span />
  <EndPoint
    path={
      Object {
        "test2": "test2",
      }
    }
    selectedServer=""
    specType="rest"
  >
     
  </EndPoint>
</Fragment>
`;
